### PR TITLE
BUGFIX: Use env var for FLOW_LOCK_HOLDING_PAGE

### LINKED ISSUE #2798: BUG: LockManager: defined refers to a constant but FLOW_LOCKHOLDINGPAGE used as environment
### Description

The possibility of specifying the path to the lock holding page in the LockManager is currently broken.
Reference is made here to a constant that has not been defined.
Since this constant is not used anywhere else, it should certainly be referring to the environment directly here, rather than to a constant.

### Steps to Reproduce

❗  The problem can only be checked in the production context.

1. Create an environment variable FLOW_LOCKHOLDINGPAGE and point to a document provided relative to the package directory. `Application/Acme.Foo/Resources/Private/CustomLockHoldingPage.html`
2. Lock your application using the LockManager's lockSiteOrExit Method.
3. Visit the application

#### Expected behavior

The HTML document `CustomLockHoldingPage.html`that was defined via the environment is displayed.

#### Actual behavior

The `Neos.Flow/Resources/Private/Core/LockHoldingStackPage.html` document is always displayed.

### Affected Versions

This feature was introduced with version 5.2. It seems it never really got to work. 
https://github.com/neos/flow-development-collection/pull/1381

Flow: 7.3.1

### Possible solution

Neos.Flow/Classes/Core/LockManager.php Line: 57

```php
    /**
     * Initializes the manager, removing expired locks
     */
    public function __construct()
    {
        $lockPath = $this->getLockPath();
        $this->lockPathAndFilename = $lockPath . md5(FLOW_PATH_ROOT) . '_Flow.lock';
        $this->lockFlagPathAndFilename = $lockPath . md5(FLOW_PATH_ROOT) . '_FlowIsLocked';
        $this->lockHoldingPage = getenv('FLOW_LOCKHOLDINGPAGE')
            ? FLOW_PATH_PACKAGES . getenv('FLOW_LOCKHOLDINGPAGE')
            : FLOW_PATH_FLOW . 'Resources/Private/Core/LockHoldingStackPage.html';
        $this->removeExpiredLock();
    }
```

On this occasion it would be possible to rename the corresponding environment variable from `FLOW_LOCKHOLDINGPAGE` to `FLOW_LOCK_HOLDING_PAGE`.

### PR DESCRIPTION
**Pull request by kdambekalns**
The lock holding page was supposed to be customizable through
an environment variable, but that never worked…

Now it does, using the new variable name `FLOW_LOCK_HOLDING_PAGE`
as well as the (now deprecated) old name `FLOW_LOCKHOLDINGPAGE`.

Fixes #2798

### CODE REVIEW

**Comment by kdambekalns** [comment]
Seems noone ever tried to use it, really… 🙊

**Review by jonnitto — APPROVED** [review]
By reading 📖

**Comment by kdambekalns** [comment]
Thanks @jonnitto – you approved the original PR, too. 😜

File: Neos.Flow/Classes/Core/LockManager.php [hunk 1]
```diff
@@ -54,7 +54,8 @@ public function __construct()
         $lockPath = $this->getLockPath();
         $this->lockPathAndFilename = $lockPath . md5(FLOW_PATH_ROOT) . '_Flow.lock';
         $this->lockFlagPathAndFilename = $lockPath . md5(FLOW_PATH_ROOT) . '_FlowIsLocked';
-        $this->lockHoldingPage = defined('FLOW_LOCKHOLDINGPAGE') ? FLOW_PATH_PACKAGES . FLOW_LOCKHOLDINGPAGE : FLOW_PATH_FLOW . 'Resources/Private/Core/LockHoldingStackPage.html';
+        $configuredLockHoldingPage = Bootstrap::getEnvironmentConfigurationSetting('FLOW_LOCKHOLDINGPAGE');
```
Comment (albe) [inline Neos.Flow/Classes/Core/LockManager.php:57]:
```suggestion
        $configuredLockHoldingPage = Bootstrap::getEnvironmentConfigurationSetting('FLOW_LOCK_HOLDING_PAGE') ?? Bootstrap::getEnvironmentConfigurationSetting('FLOW_LOCKHOLDINGPAGE');
```
😬 ❓

Comment (albe) [reply Neos.Flow/Classes/Core/LockManager.php]: Exactly that. But we are nice and make the old name work as well :)

**Comment by albe** [comment]
What shall we do with the documentation at https://github.com/kdambekalns/flow-development-collection/blob/bugfix/env-var-lockholdingpage/Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst? Keep it like that and change it for 8.0 where we deprecate the old name and keep it for another major version?

**Comment by kdambekalns** [comment]
I'd say adjust it to the new variable name and mention the old one as deprecated. Since no-one ever used it successfully, we can get away with that. 😎

**Comment by kdambekalns** [comment]
🙄

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst [hunk 2]
```diff
@@ -190,16 +190,18 @@ Environment Variables
...
-								This is relative to the Packages directory.
-FLOW_ONLY_COMPOSER_LOADER		Set to true (1) to only use composer autoloader.
-============================	==================================================
+============================= ==================================================
+Variable                      Description
+============================= ==================================================
+``FLOW_ROOTPATH``             Can be used to override the path to the Flow root.
+``FLOW_CONTEXT``              Use to set the flow context (see above).
+``FLOW_PATH_TEMPORARY_BASE``  Can be used to set a path for temporary data.
+``FLOW_LOCKHOLDINGPAGE``      Use to specify the html page shown when the site is
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst:199]:
```suggestion
``FLOW_LOCK_HOLDING_PAGE``      Use to specify the html page shown when the site is
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst [hunk 3]
```diff
@@ -190,16 +190,18 @@ Environment Variables
...
+                              locked. This is relative to the Packages directory.
+                              Can be given as ``FLOW_LOCKHOLDINGPAGE``, too.
+                              That is deprecated as of Flow 8.0.
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst:202]: Shall we mention deprecation in Flow 8 in the 6.3+ docs? Feels weird, but I guess it doesn't hurt and makes things more transparent, so...

**Review by albe — APPROVED** [review]

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst (hunk 2 above)
Comment (kdambekalns) [reply Neos.Flow/Documentation/TheDefinitiveGuide/PartII/Configuration.rst]: I swear I had it like this… 🤔

**Pull request merged**